servers/src/grin/sync/syncer.rs

```diff
@@ -147,13 +161,13 @@ impl SyncRunner {
 			}
 
 			// if syncing is needed
-			let head = self.chain.head().unwrap();
+			let head = unwrap_or_retry!(self.chain.head());
```

Comment:
The syncer already loops and implicitly retries on next loop.
So it feels like we should just break out of this loop early if we fail any of these calls that _could_ potentially fail?

I'm not convinced we need to handle these individual failures at this level of granularity like this.

Reply:
Maybe one option would be to just kill the sync thread on any kind of (retry-able) failure and kick off a new one to replace it?

Reply:
What do you mean by granularity? I think we should handle error cases using `unwarap` as the last resort, which is quite granular.

Unwrap kills a thread silently, unless we expect it. Another option is to return `Result` from syncer and handle on the upper level, but to me it seems too complicated

Reply:
Ahh I think I was misunderstanding what `unwrap_or_retry!()` was doing. I thought it was retrying the thing being wrapped in the macro. It's actually `continue` on the outer loop.

> Another option is to return Result from syncer and handle on the upper level, but to me it seems too complicated

I think this might be easier to read though.
We spawn a simple loop that handles errors (by logging errors and sleeping).
And the loop calls `sync_once()` or something similar returning `Result`.

Reply:
> Ahh I think I was misunderstanding what unwrap_or_retry!() was doing. I thought it was retrying the thing being wrapped in the macro. It's actually continue on the outer loop.

Yeah, this wasn't obvious to me either - either follow @antiochp's suggestion or add a comment in the continue statement in the macro that it's meant to continue the sync loop.